### ISSUE TITLE
Catch all errors from sign in - improvement for signInFailure 

### ISSUE THREAD

**Issue by Boardtale** [firebase/firebaseui-web#1074]
Need:

I require for email + pwd to verify emails.
As far as I know, I can't prevent or redirect flow after user first time creates account. So the error is sent from my blocking function: 'please verify email'.
That does not look good when user tries to log in with unverfied email:
![image](https://github.com/firebase/firebaseui-web/assets/132870043/12b3e553-e793-4664-9f77-d01b2c40ca4b)
I want to catch that error and show proper error popup for the users.

I am using 'signInFailure' but it's not triggered.. I think ever :P I don't know where it's triggered but not if I get any forbidden from my blocking function

So I'd like to solve it somehow. I see few paths but neither is possible - afaik - with firebaseUI. Unless I'm gonna dirty trick it and break encapsulation.

1. Stop auto sign in after creation of the account for email. I could redirect user in such case to dedicated page where I ask him to verify account. Also I need to be able to send that email there! So it's important for me to get handler
2. Ignore the flow, but be able to catch all errors. Some kind of generic catch for all that thing.

**Comment by Thanush19** [comment]
i can fix this @Boardtale 
can i work on this??

**Comment by jhuleatt** [comment]
Hi @Boardtale, thanks for reporting this. I agree that `signInFailure` should trigger if account creation or sign-in is blocked by a blocking function. We'll mark this as a bug.

As a side note, would [email link authentication](https://github.com/firebase/firebaseui-web?tab=readme-ov-file#email-link-authentication) be a better fit for your use case than email + password?

**Comment by hungrydaddy** [comment]
Glad that I can confirm that I am not the only one experiencing this.
My case is really similar where I am blocking user creation in the beforeCreate blocking function, returning a custom error that I prepare to catch and handle on the frontend to display a user-friendly error.
Unfortunately I am in the same boat here @Boardtale , there seems no way to catch this error anywhere.

The only potential approach for me now is to use firebase SDK to initiate sign in flow which then defeats the purpose of me using this library.

**Comment by dackers86** [comment]
Thank you for taking the time to submit this issue.

A new release is planned shortly, which should resolve this request. 

In the meantime, we will close this  bug, if this is not resolved in the new version - please feel free to create a new issue bug/feature request.
